.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -0,0 +1,77 @@
+name: Docker Build & Publish
+
+on:
+  workflow_call:
+    inputs:
+      dockerfile:
+        required: false
+        type: string
+        description: "Path to Dockerfile"
+        default: "Dockerfile"
+      dockerContext:
+        required: false
+        type: string
+        description: "The Docker context"
+        default: "."
+      publish:
+        required: true
+        type: boolean
+
+jobs:
+  docker_build:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        if: inputs.publish == true
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Determine image name
+        id: set_image_name
+        run: echo "IMAGE_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)" >> $GITHUB_ENV
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          tags: ${{ env.IMAGE_NAME }}:${{ github.sha }}
```

Comment:
Can we build with both SHA and tag if existing?
This will require updates on the push as well.
CircleCI ref: https://github.com/babylonchain/staking-indexer/blob/0e32c9d8c749c5c282e3e17ea5aff02756fa1911/.circleci/config.yml#L56

Reply:
Added